hal/inc/hal_dynalib_core.h

```diff
@@ -90,6 +91,23 @@ DYNALIB_FN(40, hal_core, hal_get_device_secret, int(char*, size_t, void*))
 DYNALIB_FN(41, hal_core, hal_set_device_secret, int(char*, size_t, void*))
 DYNALIB_FN(42, hal_core, hal_get_device_hw_info, int(hal_device_hw_info*, void*))
 
+#define BASE_IDX 43
+
+#if HAL_PLATFORM_HW_WATCHDOG
+DYNALIB_FN(BASE_IDX + 0, hal_core, hal_watchdog_set_config, int(hal_watchdog_instance_t, const hal_watchdog_config_t*, void*))
```

Comment:
Perhaps let's also move these into a separate dynalib table, it doesn't make sense to have so many watchdog related calls now in core hal.